tools/internal/autoupdate/githublatestrelease.go

```diff
@@ -5,31 +5,90 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/rancher/image-mirror/internal/config"
-
+	"github.com/Masterminds/semver/v3"
 	"github.com/google/go-github/v71/github"
+	"github.com/rancher/image-mirror/internal/config"
 )
 
-// GithubLatestRelease retrieves the tag of the latest release
-// of the configured github repository, and returns the configured
-// Images with this tag. It assumes that the images have the same
-// tags as the github releases.
-type GithubLatestRelease struct {
-	Owner      string
-	Repository string
-	Images     []AutoupdateImageRef
+// GithubRelease retrieves the tags of all github releases that match the VersionConstraint if
+// LatestOnly is false, or the tag of the latest release if LatestOnly is true.
+// It returns the configured Images with this tag.
+// It assumes that the images have the same tags as the github releases.
+type GithubRelease struct {
+	Owner             string
+	Repository        string
+	Images            []AutoupdateImageRef
+	LatestOnly        bool
+	VersionConstraint string
 }
 
-func (glr *GithubLatestRelease) GetUpdateImages() ([]*config.Image, error) {
+func (gr *GithubRelease) GetUpdateImages() ([]*config.Image, error) {
 	client := github.NewClient(nil)
-	release, _, err := client.Repositories.GetLatestRelease(context.Background(), glr.Owner, glr.Repository)
+	if gr.LatestOnly {
+		return gr.getImagesFromLatestRelease(client)
+	} else {
+		return gr.getImagesFromAllReleases(client)
+	}
+}
+
+func (gr *GithubRelease) getImagesFromAllReleases(client *github.Client) ([]*config.Image, error) {
+	opt := &github.ListOptions{}
+
+	images := make([]*config.Image, 0, len(gr.Images))
+	for {
+		releases, resp, err := client.Repositories.ListReleases(context.Background(), gr.Owner, gr.Repository, opt)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get releases: %w", err)
+		}
+
+		for _, release := range releases {
+			if release.GetPrerelease() || release.GetDraft() {
+				continue
+			}
+			if gr.VersionConstraint != "" {
+				version, err := semver.NewVersion(release.GetTagName())
+				if err != nil {
+					return nil, fmt.Errorf("error parsing release version: %w", err)
+				}
+
+				constraint, err := semver.NewConstraint(gr.VersionConstraint)
```

Comment:
nit: since there are other comments we should address, we might as well put this outside of the nested for loop.